src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentState.cs

```diff
@@ -235,11 +235,8 @@ private VersionStamp GetNewerVersion()
             return VersionStamp.Create();
         }
 
-        public virtual async Task<VersionStamp> GetTopLevelChangeTextVersionAsync(CancellationToken cancellationToken)
-        {
-            var textAndVersion = await this.TextAndVersionSource.GetValueAsync(LoadTextOptions, cancellationToken).ConfigureAwait(false);
-            return textAndVersion.Version;
-        }
+        public virtual Task<VersionStamp> GetTopLevelChangeTextVersionAsync(CancellationToken cancellationToken)
+            => this.TextAndVersionSource.GetVersionAsync(LoadTextOptions, cancellationToken);
```

Comment:
this code now calls through to the fastpath to avoid loading text just to get version info.